src/libraries/System.Private.CoreLib/src/System/Buffers/Text/FormattingHelpers.CountDigits.cs

```diff
@@ -24,8 +24,9 @@ public static int CountDigits(ulong value)
             ];
             Debug.Assert(log2ToPow10.Length == 64);
 
+            // We use 'nint' because it gives us a free early zero-extension to 64 bits when running on a 64-bit platform.
             // TODO: Replace with log2ToPow10[BitOperations.Log2(value)] once https://github.com/dotnet/runtime/issues/79257 is fixed
-            uint index = Unsafe.Add(ref MemoryMarshal.GetReference(log2ToPow10), BitOperations.Log2(value));
+            nint elementOffset = Unsafe.Add(ref MemoryMarshal.GetReference(log2ToPow10), BitOperations.Log2(value));
```

Comment:
why `nint` isn't a sign rather than zero extension. Also, this looks like a hack that JIT should do, is it really needed?

Reply:
have you checked that it comes with codegen improvements? if not (or it's not clear whether it brings any benefits) let's revert it. We don't want to pollute our codebase with hacks which often becomes unnecessary eventually.

Reply:
or at least remove the comment

Reply:
Using early-extension does provide a code generation improvement, e.g.

```diff
; Assembly listing for method System.Int128:TryFormat(System.Span`1[ushort],byref,System.ReadOnlySpan`1[ushort],System.IFormatProvider):ubyte:this (FullOpts)
        mov      rdx, 0xD1FFAB1E      ; static handle
        movzx    rdi, byte  ptr [rsi+rdx]
-       mov      esi, edi
-       mov      rdx, 0xD1FFAB1E      ; static handle
-       cmp      rax, qword ptr [rdx+8*rsi]
+       mov      rsi, 0xD1FFAB1E      ; static handle
+       cmp      rax, qword ptr [rsi+8*rdi]
```

But it is not dependant on using `nint` vs `nuint`.